SU-SWS/linky_clicky
# Basic 2385/caleb

# READY FOR REVIEW
@jbickar I ran and fixed, if necessary, all the tests I could. It's ready for your review.

# Summary
- Caleb:
 Refactored all of the Jumpstart product features to not use Scenario Outlines. All of the Scenario Outline pieces are written out as individual steps.

A few of the commit messages were labelled as the incorrect product, namely:
  - stanford_person_layout_views.feature labelled as jsa is actually in jse
  - The second homepage.feature labelled as jsa is actually in jsl

# Needed By (Date)
- When possible

# Criticality
- Fixes slow scenarios - Will make Behat tests run faster

# Steps to Test
## 1. Switch to this branch
## 2. Test  stanford_jumpstart_home 
For JSA, start from `linky_clicky/products/jsa`
`../../bin/behat -p sites-sws-build-jsa -s dev features/stanford_jumpstart_home`

For JSE start from: `linky_clicky/products/jse`
`./../bin/behat -p sites-sws-build-jse -s dev features/stanford_jumpstart_home`

## 3. Test stanford_person
- Edit Haley's profile and change Affiliation from *Student* to *Students*. See [bug report](https://stanfordits.atlassian.net/browse/BASIC-2415)

1. Login to the site
2. Go to people/all/grid/grouped
2. Click on `edit` for Haley
4. Change Affiliation  from *Student* to *Students*

For JSA, From `linky_clicky/products/jse`
`../../bin/behat -p sites-sws-build-jse -s dev features/stanford_person`

For JSE From `linky_clicky/products/jsa`
`../../bin/behat -p sites-sws-build-jsa -s dev features/stanford_person`

## 4. Test stanford_sites_helper/stanford_helper/stanford_sites_helper.feature
_You'll need a brand new site.... I did not test this._

## 5. Test jsa/homepage.feature
From `linky_clicky/products/jsa`
../../bin/behat -p sites-sws-build-jsa -s dev features/homepage.feature

## 6. Test jsl/features/homepage.feature
_I don't know of a JSL site. I did not test this._

##  7. Test jslbre/features/homepage.feature
_I don't know of a JSLBRE site. I did not test this._

## 8. Test jsplus/features/homepage.feature
From: linky_clicky/products/jsplus
../../bin/behat -p sites-sws-build-jsplus -s dev features/homepage.feature

## 9. Test jsv/features/homepage.feature
From: linky_clicky/products/jsv
../../bin/behat -p sites-sws-build-jsv -s dev features/homepage.feature

## 10. Test vpsa/features/homepage.feature
From: linky_clicky/products/jsvpsa
../../bin/behat -p sites-sws-build-jsvpsa -s dev features/homepage.feature

# Affects 
- Most products especially JSE and JSA

# Associated Issues and/or People
## Related JIRA ticket(s)
https://stanfordits.atlassian.net/browse/BASIC-2385 

## Related PRs

## More Information
The student view is broken. See https://stanfordits.atlassian.net/browse/BASIC-2415


###  Example JSA profile (in behat.local.yml)

```
sites-sws-build-jsa:
  extensions:
    Behat\MinkExtension:
      base_url: https://sws-build-jsa.stanford.edu
    Drupal\DrupalExtension:
      drush:
        alias: sse.sws-build-jsa
```
### Example JSE profile (in behat.local.yml)

```
sites-sws-build-jse:
  extensions:
    Behat\MinkExtension:
      base_url: https://sites.stanford.edu/sws-build-jse
    Drupal\DrupalExtension:
      drush:
        alias: sse.sws-build-jse
```

## Folks to notify

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
